lib/cli-engine.js

```diff
@@ -754,8 +755,7 @@ CLIEngine.prototype = {
         if (filename && !isAbsolute(filename)) {
             filename = path.resolve(options.cwd, filename);
         }
-        if (filename && ignoredPaths.contains(filename)) {
-
+        if (filename && warnIgnored !== false && ignoredPaths.contains(filename)) {
```

Comment:
This works the opposite of how the `executeOnFile` version works. We always have optional arguments default to `false`, so we have to have that here as well. A bit concerned that this would result in a breaking change though. :-/